aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,394 @@
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<RetryHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+    .AddHttpMessageHandler<RetryHandler>(); // Retry requests to GitHub using the retry handler
+}
+```
+
+In the preceding code, the `RetryHandler` is registered as a transient service with DI. Once registered, `AddHttpMessageHandler` can be called, passing in the type for the handler.
+
+Multiple handlers can be registered in the order that they should execute. Each handler wraps the next handler until the final `HttpClientHandler` executes the request.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<OuterHandler>();
+    services.AddTransient<InnerHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+    .AddHttpMessageHandler<OuterHandler>() // This handler is on the outside and executes first on the way out and last on the way in.
+    .AddHttpMessageHandler<InnerHandler>(); // This handler is on the inside, closest to the request.
+}
+```
+
+## Handling errors with Polly
+
+_NOTE: This is a WIP since the Polly Extensions API is still currently under design._
+
+The preceding example demonstrated building a simple retry handler manually. A more robust and feature-rich approach is to leverage a popular third-party library called [Polly](https://github.com/App-vNext/Polly).
+
+Polly is a comprehensive resilience and transient-fault-handling library for .NET which allows developers to express policies such as Retry, Circuit Breaker, Timeout, Bulkhead Isolation, and Fallback in a fluent and thread-safe manner.
+
+Extension methods are provided for `HttpClientFactory`, which enable easy integration and use of Polly policies with configured `HttpClient` instances. Rather than manually defining a handler, extension methods can be used which accept a Polly policy. This policy is used when executing HTTP requests.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<OuterHandler>();
+    services.AddTransient<InnerHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+
+    // Build a totally custom policy using any criteria
+    .AddPolicyHandler(Policy.Handle<HttpRequestException>().RetryAsync())
+
+    // Build a policy that handles exceptions (connection failures)
+    .AddExceptionPolicyHandler(p => p.RetryAsync())
+
+    // Build a policy that handles exceptions and 500s from the remote server
+    .AddServerErrorPolicyHandler(p => p.RetryAsync())
+
+    // Build a policy that handles exceptions, 400s, and 500s from the remote server
+    .AddBadRequestPolicyHandler(p => p.RetryAsync());
+}
+```
+
+## DNS and handler rotation
```

Comment:
I think it would be great to call this section 'HttpClient' and lifetime management and introduce some of these concepts a bit more gently.